### PR TITLE
Extract inclusion of CSS from component into preview

### PR DESCRIPTION
**Pull request by romidane**
* This is to separate concerns with styling hierarchy. This is only needed in the preview so it should live in the preview.
* Users can import the styles in to their app seperatly unless the builder of the component wants specific styles to be hotloaded to the component when it mounts

### CODE REVIEW

**Comment by joshnesbitt** [comment]
Good shout. This reenforces a conversation I had with @mackstar about the UI toolkit being separate from the component. I think if you want to import the toolkit into the projects SCSS file then that's great, but it should be up to the developer integrating the component as to whether to pull the styles in or not. Happy with that @mackstar?

File: src/project-creation/preview-creator.js [hunk 1]
```diff
@@ -11,6 +11,10 @@ export default {
 import ReactDOM from 'react-dom';
 import ${klass} from './components/${name}';
 
+if (process.env.WEBPACK_BUILD) {
```
Comment (joshnesbitt) [inline src/project-creation/preview-creator.js]: I don't think we need the conditional in this case, we'll always want the styles?

Comment (romidane) [inline src/project-creation/preview-creator.js]: That's true you'd alway want the CSS 👍

**Comment by romidane** [comment]
@joshnesbitt  I don't think importing the toolkit into the component is a smart thing. There are two main reasons for this.
* The version used in the component may be different to the one used in the users project.
* The second is that once this component is included into a project that is already using toolkit then there will be twice the amount of css added to the app as there will be 2 toolkits loaded into an app which could create conflicts between the two toolkits

I think every component should have a disclaimer that they require your project to have toolkit to be installed as well as specify a minimum version if needed.

**Comment by joshnesbitt** [comment]
I'm not completely against that, I agree the duplicated CSS isn't ideal. I was chatting to @aaronthomas about this the other day. I think there should be the following options:

* If you wish, require the styles from the component, which may `@import` the toolkit and other styles custom the component. Some components may require styling to work properly (such as an autocomplete text field for example) therefore we should encourage this. If we're using the toolkit properly, we should have a granular, isolated approach to importing small chunks of UI functionality
* If you want to override the styles provided by the component, you simply don't require the styles provided by that component and instead override them in the parent application

That way we have a route which allows the component to "just work" as well as a more flexible option.

**Comment by mackstar** [comment]
Maybe it should be a `devDependency`?

**Comment by joshnesbitt** [comment]
The UI toolkit @mackstar?

**Comment by romidane** [comment]
@joshnesbitt  Hmm. I can see the dilemma here. I guess you've go a good point about having components that "just work". I guess it's all about finding the right trade off. I guess if we get a few components going and used we could get some useful feed back from the stakeholders?

**Comment by mackstar** [comment]
@joshnesbitt Yes..

For now I don't see much way around making it a dependency of the component itself.. I don't see much of a problem either, especially as since now it is on NPM... This will do version resolution for us.

**Comment by joshnesbitt** [comment]
I think it's fine to have the UI toolkit as a hard dependency in a component if the styling depends on it. As I said, I think if we're building things as granular components, this shouldn't be a problem.

**Comment by aaronthomas** [comment]
Having just had a chat with @joshnesbitt, I agree it makes sense for the Toolkit to be a hard dependency of the component, as long as the component doesn't import `_all.scss` which will cause heavy duplication – instead it can import Toolkit Core's `_tools.scss` which provides access to all non-compiling Sass, like mixins, vars, functions etc. without duping the resets and base styles.

For UI CSS – for now it makes sense to allow the duping. The components are granular enough that bloat shouldn't be a major issue, particularly in the short term while there are very few components.

If and when it becomes a bigger issue in the future and apps have numerous Blackjack dependencies all importing the same UI styles, we may need a more elegant solution...

File: src/project-creation/preview-creator.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ export default {
     return `import React from 'react';
 import ReactDOM from 'react-dom';
 import ${klass} from './components/${name}';
+import Styles from '../styles/components/${name}.scss';
```
Comment (joshnesbitt) [inline src/project-creation/preview-creator.js]: Looking at this, the linter will moan as `Styles` is never used. Never been keen at the way webpack just bungs this in the `HEAD`, but for now a require might be better. I'll tweak.

**Comment by joshnesbitt** [comment]
Thanks @aaronthomas. Are we happy with this approach? Will merge this PR if so and this can be an ongoing discussion as we integrate more components.

**Comment by romidane** [comment]
@joshnesbitt I'm happy with this approach. I think we will need to document this somewhere. So that people know how to include the toolkit in to their components 👍

**Comment by joshnesbitt** [comment]
@romidane agree, I think we'll end up creating guides at some point.

**Pull request merged**